### ISSUE TITLE
`terminal.shortcuts`: DEFAULT_BUFFER is not defined

### ISSUE THREAD

**Issue by max-moser** [ipython/ipython#13967]
I'm using IPython as the interactive Flask shell for my daily work, and it usually works like a charm.
For a little while now, I keep getting error messages while I'm jumping around in my command history (`^R` and co.) like the following:
```
Unhandled exception in event loop:
  File "/home/mmoser/.pyenv/versions/3.9.15/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/prompt_toolkit/input/vt100.py", line 173, in callback_wrapper
    callback()
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/prompt_toolkit/application/application.py", line 698, in read_from_input
    self.key_processor.process_keys()
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_processor.py", line 272, in process_keys
    self._process_coroutine.send(key_press)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_processor.py", line 187, in _process
    self._call_handler(matches[-1], key_sequence=buffer[:])
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_processor.py", line 322, in _call_handler
    handler.call(event)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/prompt_toolkit/key_binding/key_bindings.py", line 126, in call
    result = self.handler(event)
  File "/home/mmoser/.local/share/virtualenvs/my-site-OggevyoG/lib/python3.9/site-packages/IPython/terminal/shortcuts/__init__.py", line 468, in reset_search_buffer
    event.app.layout.focus(DEFAULT_BUFFER)

Exception name 'DEFAULT_BUFFER' is not defined
Press ENTER to continue...
```

My currently installed `ipython` version is 8.11.0, and Python is 3.9.15.

Indeed, on the latest commit on the `main` branch, it looks like [`DEFAULT_BUFFER` is used but never defined](https://github.com/ipython/ipython/blob/main/IPython/terminal/shortcuts/__init__.py#L468).

**Comment by krassowski** [comment]
Thank you for the report, and apologies for the inconvenience - I forgot to move this import when refactoring shortcuts to enable customisation and this code branch was not covered by tests. There is already a PR fixing this (https://github.com/ipython/ipython/pull/13961).

**Comment by max-moser** [comment]
Wow, thanks for the quick action!
And thanks for making the Python REPL so much more usable!

**Comment by Carreau** [comment]
You need to update IPython, you likely have IPIython 8.11 or earlier, it is fixed in 8.12 or later.

**Comment by tmzhuang** [comment]
Ah makes sense. Thanks for the quick response!
